src/localplayer.cpp

```diff
@@ -1077,3 +1074,67 @@ float LocalPlayer::getSlipFactor(Environment *env, const v3f &speedH)
 	}
 	return 1.0f;
 }
+
+void LocalPlayer::handleAutojump(f32 dtime, Environment *env,
+		const collisionMoveResult &result, const v3f &old_position,
+		const v3f &old_speed, f32 pos_max_d)
+{
+	PlayerSettings &player_settings = getPlayerSettings();
+	bool control_forward = control.up ||
+			       (!control.up && !control.down &&
+					       control.forw_move_joystick_axis < -0.05);
+	if (autojump) {
+		// release autojump after a given time
+		autojump_time -= dtime;
+		if (autojump_time <= 0.0f) {
+			autojump = false;
+		}
+	} else if (m_can_jump && !control.jump && !control.sneak && control_forward &&
```

Comment:
Invert and return. Keeps indents to a minimum and makes the code easier to read (because no `else` case is to expect).

Reply:
Refactored the entire method, much less indentation now.